tests/node_test.cc

```diff
@@ -1201,5 +1285,94 @@ TEST_CASE("Node is checked for 3D case", "[node][3D]") {
         REQUIRE(node->acceleration(Nphase)(i) ==
                 Approx(acceleration(i)).epsilon(Tolerance));
     }
+
+    SECTION("Check general velocity constraints") {
+      // Check momentum
+      Eigen::Matrix<double, Dim, 1> momentum;
+      for (unsigned i = 0; i < momentum.size(); ++i) momentum(i) = 10.;
+
+      // Check initial momentum
+      for (unsigned i = 0; i < momentum.size(); ++i)
+        REQUIRE(node->momentum(Nphase)(i) == Approx(0.).epsilon(Tolerance));
+
+      // Check assign momentum to 10
+      REQUIRE(node->update_momentum(false, Nphase, momentum) == true);
+      for (unsigned i = 0; i < momentum.size(); ++i)
+        REQUIRE(node->momentum(Nphase)(i) == Approx(10.).epsilon(Tolerance));
+
+      // Check zero velocity
+      for (unsigned i = 0; i < Dim; ++i)
+        REQUIRE(node->velocity(Nphase)(i) == Approx(0.).epsilon(Tolerance));
+
+      // Check mass
+      mass = 100.;
+      node->update_mass(false, Nphase, mass);
+      REQUIRE(node->mass(Nphase) == Approx(100.).epsilon(Tolerance));
+
+      // Compute and check velocity
+      node->compute_velocity();
+      for (unsigned i = 0; i < Dim; ++i)
+        REQUIRE(node->velocity(Nphase)(i) == Approx(0.1).epsilon(Tolerance));
+
+      // Check acceleration
+      Eigen::Matrix<double, Dim, 1> acceleration;
+      for (unsigned i = 0; i < acceleration.size(); ++i) acceleration(i) = 5.;
+      REQUIRE(node->update_acceleration(false, Nphase, acceleration) == true);
+      for (unsigned i = 0; i < acceleration.size(); ++i)
+        REQUIRE(node->acceleration(Nphase)(i) == Approx(5.).epsilon(Tolerance));
+
+      // Check velocity before constraints
+      Eigen::Matrix<double, Dim, 1> velocity;
+      velocity << 0.1, 0.1, 0.1;
+      for (unsigned i = 0; i < velocity.size(); ++i)
+        REQUIRE(node->velocity(Nphase)(i) ==
+                Approx(velocity(i)).epsilon(Tolerance));
+
+      // Check acceleration before constraints
+      acceleration << 5., 5., 5.;
+      for (unsigned i = 0; i < acceleration.size(); ++i)
+        REQUIRE(node->acceleration(Nphase)(i) ==
+                Approx(acceleration(i)).epsilon(Tolerance));
+
+      // Apply velocity constraints
+      REQUIRE(node->assign_velocity_constraint(0, 10.5) == true);
+      REQUIRE(node->assign_velocity_constraint(1, -12.5) == true);
```

Comment:
These are not sufficient tests as we are not testing for the third direction. Ideally I'd at least test 2 cases: One with the third dimension unaltered like this case, and one with the constraint in the third dimension.

Reply:
I did:
- 1 direction constraint for 2D
- 2 directions constraint for 2D
- 2 directions constraint for 3D
- 3 directions constraint for 3D